### ISSUE TITLE
have a Trouble at ios device

### ISSUE THREAD

**Issue by RyanTse1** [facebookresearch/segment-anything#520]
when i select modle area   at  IOS device， the response is so slowly (usually 5 sec) ,    but  in  android / windows devices ( 1-2 sec) 

sam  has some compatibility issues    with  IOS  system ? 

anyone  meet  this problem  ?

**Comment by ryouchinsa** [comment]
Here is the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos/tree/master).
This code is originated from [Segment Anything CPP Wrapper](https://github.com/dinglufe/segment-anything-cpp-wrapper) and implemented on macOS app [RectLabel](https://rectlabel.com). We customized the original code so that getMask() uses the previous mask result called as low_res_logits and retain the previous mask array for undo/redo actions. 

We hope this macOS version would help to develop the iOS version.
Please let us know your opinion.

![sam_polygon](https://github.com/ryouchinsa/sam-cpp-macos/assets/1954306/4640e139-c533-4b8c-b27b-e02a401b9bbd)

**Comment by ryouchinsa** [comment]
Improved "Create polygon using SAM" feature so that you can label pixels using the pixels option. You can label [Segment Anything 1 Billion (SA-1B)](https://ai.meta.com/datasets/segment-anything-downloads/) like dataset by yourself.

![sam-pixels](https://github.com/facebookresearch/segment-anything/assets/1954306/d828c18b-6fda-4258-b866-21bc34a0d17f)

**Comment by sobamemil** [comment]
Did you converted SAM to MLModel?

**Comment by ryouchinsa** [comment]
We are running ONNX models using the [Segment Anything CPP Wrapper for macOS](https://github.com/ryouchinsa/sam-cpp-macos#segment-anything-cpp-wrapper-for-macos).

**Comment by roimulia2** [comment]
Can this be used on iOS natively? I assume it'll be to slow for real-time, but for a single instance image, do you find it capable enough?
